components/CharmCalculator.tsx

```diff
@@ -0,0 +1,198 @@
+"use client";
+
+import { useState } from "react";
+
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
+import { Table, TableBody, TableCaption, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+
+import Heroes from "../public/data/heroes.json";
+
+import { Charm, CharmPosition, StatSummary, GetStatSummary } from "@/lib/charm";
+
+export default function CharmCalculator() {
+  const availableClasses = Heroes.Classes.map((classType) => classType.Class);
+  const [selectedClass, setSelectedClass] = useState<String>(availableClasses[0]);
+
+  const [availableHeroes, setAvailableHeroes] = useState<any[] | undefined>(undefined);
```

Comment:
init this with the list of heroes instead of having the optional `undefined` type